pkg/domain/infra/abi/play.go

```diff
@@ -1366,6 +1367,15 @@ func (ic *ContainerEngine) PlayKubeDown(ctx context.Context, body io.Reader, opt
 		}
 	}
 
+	// Remove the service container to ensure it is removed before we return for the remote case
+	// Needed for the clean up with podman kube play --wait in the remote case
+	if reports.ServiceContainerID != "" {
+		_, err = ic.ContainerRm(ctx, []string{reports.ServiceContainerID}, entities.RmOptions{})
+		if err != nil && !errorhandling.Contains(err, define.ErrNoSuchCtr) {
```

Comment:
You can use `errors.Is()` here since we're not going over the wire here.

Reply:
fixed